compiler/core/lam.ml

```diff
@@ -719,8 +715,6 @@ let result_wrap loc (result_type : External_ffi_types.return_wrapper) result =
   | Return_null_to_opt -> prim ~primitive:Pnull_to_opt ~args:[result] loc
   | Return_null_undefined_to_opt ->
     prim ~primitive:Pnull_undefined_to_opt ~args:[result] loc
-  | Return_undefined_to_opt ->
```

Comment:
Removing `@return(undefined_to_opt)` sounds good, as it does not really have a clear use case.